src/tz.cpp

```diff
@@ -2121,22 +2126,24 @@ time_zone::init() const
 }
 
 sys_info
-time_zone::load_sys_info(std::vector<detail::transition>::const_iterator i) const
+time_zone::load_sys_info(std::vector<detail::transition>::const_iterator i, sys_seconds tp) const
 {
+    auto load_posix_info = [&](){auto tz = Posix::time_zone{ posix_timezone_ }.get_info(tp); if (tz.begin < transitions_.back().timepoint){tz.begin = transitions_.back().timepoint;} return tz;};
```

Comment:
To reduce line changes define a lambda that handles fetching Posix Info correctly. Its an ugly one liner but just links the posix timezone and the final timezone transition together.